hack/jenkins/test-history/gen_html.py

```diff
@@ -143,36 +145,67 @@ def gen_html(data, prefix, exact_match=False):
+def gen_metadata_links(suites):
+    html = []
+    for (name, target), (okay, unstable, failed, skipped) in sorted(suites.iteritems()):
+        html.append('<a class="suite-link" href="{}">'.format(target))
         html.append('<span class="total okay">{}</span>'.format(okay))
         html.append('<span class="total unstable">{}</span>'.format(unstable))
         html.append('<span class="total failed">{}</span>'.format(failed))
-        html.append(suite)
+        html.append(name)
         html.append('</a>')
-    html.append('</body>')
-    html.append('</html>')
-    return '\n'.join(html)
+    return html
 
-if __name__ == '__main__':
-    prefix = ''
-    if len(sys.argv) == 2:
-        prefix = sys.argv[1]
-    with open('tests.json') as f:
+def main(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--suites', action='store_true',
+                        help='output test results for each suite')
+    parser.add_argument('--prefixes',
+                        help='comma-separated list of suite prefixes to create pages for')
+    parser.add_argument('--output-dir', required=True,
+                        help='where to write output pages')
+    parser.add_argument('--input', required=True,
+                        help='JSON test data to read for input')
+    options=parser.parse_args(args)
+
+    with open(options.input) as f:
         data = json.load(f)
-    if len(sys.argv) == 3:
-        assert sys.argv[1] == 'gensuites'
+
+    if options.prefixes:
+        # the empty prefix means "all tests"
+        options.prefixes = [''] + options.prefixes.split(',')
+        prefix_metadata = {}
+        for prefix in options.prefixes:
+            if prefix:
+                path = 'tests-%s.html' % prefix
+                prefix = 'kubernetes-' + prefix
+            else:
+                path = 'tests.html'
+            html, prefix_metadata[prefix or 'kubernetes', path] = gen_html(data, prefix, False)
+            with open(options.output_dir + '/' + path, 'w') as f:
+                f.write(html)
+    if options.suites:
         suites_set = set()
-        suite_metadata = {}
         for test, suites in data.iteritems():
             suites_set.update(suites.keys())
+        suite_metadata = {}
         for suite in sorted(suites_set):
-            html, suite_metadata[suite] = gen_html(data, suite, True)
-            with open(sys.argv[2] + 'suite-' + suite + '.html', 'w') as f:
+            path = 'suite-' + suite + '.html'
+            html, suite_metadata[suite, path] = gen_html(data, suite, True)
+            with open(options.output_dir + '/' + path, 'w') as f:
                 f.write(html)
-        with open(sys.argv[2] + 'suite-overview.html', 'w') as f:
-            f.write(gen_suite_overview(suite_metadata))
-    else:
-        print(gen_html(data, prefix, False)[0])
+    html = html_header()
+    html.append('<h1>Kubernetes Tests</h1>')
+    html.append('Last updated ' + time.strftime('%F'))
```

Comment:
Style guide says `x = a + b` is fine. I'll clean up the cases of `x = a + b + c`